src/mozanalysis/experiment.py

```diff
@@ -0,0 +1,545 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+from pyspark.sql import functions as F
+
+from mozanalysis.utils import add_days
+
+
+class Experiment(object):
+    """Get DataFrames of experiment data; store experiment metadata.
+
+    The methods here query data in a way compatible with the following
+    principles, which are important for experiment analysis:
+
+    - The population of clients in each branch must have the same
+        properties, aside from the intervention itself and its
```

Comment:
we might not need to do so here but we, in general, use inconsistent terms. for example, we say intervention here but treatment is used elsewhere (branch names in experiments). we should unify how we discuss these terms

Reply:
I whipped up a [gist](https://gist.github.com/felixlawrence/5d67e7a5beafae32a89ed2eada794b98) and posted it to the team slack. I think this conversation is outside the scope of this PR. Are you happy enough with the language here that you'd be OK with this being merged, with the understanding that we can change it later if we decide on different language?

Reply:
For posterity, the gist became a [gdoc](https://docs.google.com/document/d/16SGUwosOfwY6-eaxPj1ZyoWjJU0BZLFroQVPSnUdGoI/).

Reply:
can you put this doc in our Team google drive? eventually this should be attached to our Onboarding doc, but great to see this start

Reply:
It's [already there](https://drive.google.com/drive/folders/1LIVi1MQ58VLCMfkRy34pf-j1GS60HbFB)